src/nvim/tag.c

```diff
@@ -2782,8 +2782,7 @@ int get_tags(list_T *list, char_u *pat)
       TAG_REGEXP | TAG_NOIC, (int)MAXCOL, NULL);
   if (ret == OK && num_matches > 0) {
     for (i = 0; i < num_matches; ++i) {
-      int parse_result = parse_match(matches[i], &tp);
-      assert(parse_result == OK);
+      assert(parse_match(matches[i], &tp) == OK);
```

Comment:
We could use this sort of solution here instead:

```c
      int parse_result = parse_match(matches[i], &tp);
      (void) parse_result;
      assert(parse_result == OK);
```

That avoids the warning, but works correctly.  Perhaps some of these can be wrapped with macros that do something similar?  Any thoughts?